test/integration/integration.py

```diff
@@ -17,39 +17,53 @@
 
 class TestIntegration(unittest.TestCase):
     def test_integration(self):
-        rospy.logdebug("Integration Test Starting")
+        rospy.logdebug("Autonomy integration testing starting...")
 
         rospy.init_node("integration_test")
 
-        rospy.loginfo("Integration Test Ready")
+        rospy.loginfo("Ready")
 
-        waypoint_in_world = SE3(position=np.array([-5.5, -5.5, 0.0]))
         waypoints = [
             (
                 Waypoint(tag_id=0, type=WaypointType(val=WaypointType.NO_SEARCH)),
-                waypoint_in_world,
+                SE3(position=np.array([4, 4, 0])),
             ),
+            (
+                Waypoint(tag_id=0, type=WaypointType(val=WaypointType.POST)),
+                SE3(position=np.array([-2, -2, 0])),
+            ),
+            (
+                Waypoint(tag_id=1, type=WaypointType(val=WaypointType.POST)),
+                SE3(position=np.array([11, -10, 0])),
+            )
         ]
-
-        publish_waypoints([convert_waypoint_to_gps(waypoint) for waypoint in waypoints])
+        publish_waypoints(list(map(convert_waypoint_to_gps, waypoints)))
 
         tf_buffer = tf2_ros.Buffer()
         tf2_ros.TransformListener(tf_buffer)
 
-        has_reached_target = False
-
         while not rospy.is_shutdown():
-            try:
-                rover_in_world = SE3.from_tf_tree(tf_buffer, parent_frame="map", child_frame="base_link")
-                distance_to_target = waypoint_in_world.pos_distance_to(rover_in_world)
-                rospy.logdebug(distance_to_target)
-                if distance_to_target < COMPLETION_TOLERANCE:
-                    has_reached_target = True
-                if has_reached_target:
-                    rospy.signal_shutdown("Finished test")
-                    break
-            except tf2_ros.TransformException as exception:
-                pass
+            rate = rospy.Rate(10)
+            for waypoint in waypoints:
+                rospy.loginfo(f"Moving to waypoint: {waypoint}")
+
+                def distance_to_current_waypoint():
```

Comment:
somewhere in here we should add a ground truth check to verify localization is accurate, I can do it in this pr if you want or it can be a later feature

Reply:
Yeah that would be good. I also think we should check nav_state. Or better yet, we can launch the LED node too and check the output